### PR TITLE
Add ILIKE feature for PGVectorStore and NileVectorStore.

### PR DESCRIPTION
**Pull request by ashvin-a**
# Description

In this pull request, I have added a new `TEXT_MATCH_INSENSITIVE` operator that will handle case insensitive text matches. I have implemented this feature in PGVectorStore and NileVectorStore classes.

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-core/llama_index/core/vector_stores/types.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ class FilterOperator(str, Enum):
     ANY = "any"  # Contains any (array of strings)
     ALL = "all"  # Contains all (array of strings)
     TEXT_MATCH = "text_match"  # full text match (allows you to search for a specific substring, token or phrase within the text field)
+    ILIKE = "ilike"  # full text match (case insensitive)
```
Comment (logan-markewich) [inline llama-index-core/llama_index/core/vector_stores/types.py:78]: Not sure I like this naming, maybe `TEXT_MATCH_INSENSITIVE` is more consistent?

Comment (ashvin-a) [reply llama-index-core/llama_index/core/vector_stores/types.py:78]: Yeah that's consistent but its kinda lengthy for an operator name.

Comment (ashvin-a) [reply llama-index-core/llama_index/core/vector_stores/types.py:78]: I'll change it to `TEXT_MATCH_INSENSITIVE` right away.

Comment (ashvin-a) [reply llama-index-core/llama_index/core/vector_stores/types.py:78]: Or should I try changing `TEXT_MATCH` to `LIKE`  @logan-markewich ? What do you think?

Comment (logan-markewich) [reply llama-index-core/llama_index/core/vector_stores/types.py]: Yea its long, but tbh its more in-line. `LIKE`/`ILIKE` is very SQL-esque

**Review by logan-markewich — APPROVED** [review]
I'll have to release core before I can release the update to the individual vector stores. I should get to it tomorrow

**Comment by ashvin-a** [comment]
Some tests that are unrelated to my change has failed. What should I do?

**Comment by logan-markewich** [comment]
@ashvin-a don't worry about it :)

**Pull request merged**